### PR TITLE
Homework#1 unique_ptr

### PR DESCRIPTION
**Pull request by WojtekMs**

### CODE REVIEW

File: homework/unique_ptr/inc/unique_ptr.hpp [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+T* unique_ptr<T>::release() {
+    isOwned = false;
+    return data_;
+}
+
+template <typename T>
+void unique_ptr<T>::reset(T* new_ptr) {
+    if (isOwned) {
+        delete data_;
+    }
```
Comment (WBigaj) [inline homework/unique_ptr/inc/unique_ptr.hpp:80]: Deleting nullptr is safe.

**Review by WBigaj — COMMENTED** [review]
I have checked your code and left my feedback.

File: homework/unique_ptr/inc/unique_ptr.hpp (hunk 1 above)
Comment (MaciejSalamonski) [reply homework/unique_ptr/inc/unique_ptr.hpp:80]: +1

**Review by MaciejSalamonski — APPROVED** [review]

File: homework/unique_ptr/inc/unique_ptr.hpp (hunk 1 above)
Comment (WojtekMs) [reply homework/unique_ptr/inc/unique_ptr.hpp:80]: It's not about deleting nullptr, in the release function I am not setting data_ to nullptr. If I didn't use this boolean I would have deleted pointer that was released. I will change my implementation and remove isOwned boolean.

File: homework/unique_ptr/inc/unique_ptr.hpp [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+template <typename T>
+unique_ptr<T>& unique_ptr<T>::operator=(unique_ptr&& rhs) {
+    if (this == &rhs) {
+        return *this;
+    }
+    delete data_;
+    data_ = rhs.data_;
+    rhs.data_ = nullptr;
+    return *this;
+}
```
Comment (jakjan95) [inline homework/unique_ptr/inc/unique_ptr.hpp:40]: You can rearrange a loop to return *this just once by using ```this != &rhs``` statement

**Review by jakjan95 — APPROVED** [review]
Code looks good :) just try to change one mentioned thing and it will be perfect

File: homework/unique_ptr/tests/unique-test.cpp [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    cs::unique_ptr<int> ptr{new int{default_value}};
+    EXPECT_EQ(*ptr, default_value);
+}
+
+TEST_F(unique, arrowOperatorCanBeUsed) {
+    struct foo {
+        void use(){};
+    };
+    cs::unique_ptr<foo> ptr{new foo()};
+    ptr->use();
```
Comment (ziobron) [inline homework/unique_ptr/tests/unique-test.cpp:43]:
```suggestion
    EXPECT_NO_THROW(ptr->use());
```
You can use EXPECT_NO_THROW. Without this, in fact, this test do not test anything (or at least do not use any assert/expect) ;)

**Review by ziobron — APPROVED** [review]
Very good solution!

**Comment by ziobron** [comment]
{ "requirement_ids": [3784, 1119] }
36 XP za implementację unique_ptra (3784)
3 XP za punktualność

File: homework/unique_ptr/tests/unique-test.cpp (hunk 3 above)
Comment (WojtekMs) [reply homework/unique_ptr/tests/unique-test.cpp:43]: Thanks!

**Pull request closed without merging**